packages/modeling/src/operations/modifiers/edges.js

```diff
@@ -0,0 +1,204 @@
+const vec3 = require('../../maths/vec3')
+const line3 = require('../../maths/line3')
+
+const poly3 = require('../../geometries/poly3')
+
+// an edge is an array of two vertices
+const addEdge = (edges, edge, polygon) => {
+  let ei = edges.findIndex((element) => {
+    if (element) {
+      if (vec3.equals(element[0], edge[0]) && vec3.equals(element[1], edge[1])) return true
+      if (vec3.equals(element[0], edge[1]) && vec3.equals(element[1], edge[0])) return true
+    }
+    return false
+  })
+
+  if (ei < 0) {
+    edge.polygons = [polygon]
+    edges.push(edge)
+  } else {
+    edge = edges[ei]
+    edge.polygons.push(polygon)
+  }
+}
+
+const removeEdge = (edges, edge) => {
+  let ei = edges.findIndex((element) => {
+    if (element) {
+      if (vec3.equals(element[0], edge[0]) && vec3.equals(element[1], edge[1])) return true
+      if (vec3.equals(element[0], edge[1]) && vec3.equals(element[1], edge[0])) return true
+    }
+    return false
+  })
+  if (ei >= 0) {
```

Comment:
Should we settle on how we say "found in array". Perhaps it should always be explicitly ```ei == -1``` or ```ei != -1```

Reply:
Got it. Fixed.